File: client/mahaprinting-client/src/ServerAPI/Print.ts
import { observable } from "mobx";
import { UserPrint } from "./UserPrint";
import { PrintStatus } from "./PrintStatus";

export class Print extends UserPrint {
  @observable userId: string;

  constructor({
    id,
    name,
    fileExtension,
    slicedFor,
    status,
    timestamp,
    contactDetails,
    notes,
    userId,
  }: {
    id: number;
    name: string;
    fileExtension: string;
    slicedFor?: string;
    status: PrintStatus;
    timestamp: string;
    contactDetails: string;
    notes: string;
    userId: string;
  }) {
    super({ id, name, fileExtension, slicedFor, status, timestamp, contactDetails, notes });
    this.userId = userId;
  }
}

File: server/mahaprinting_server/DomainObjects/print.py
from enum import Enum
from typing import Optional


class PrintStatus(str, Enum):
    IN_QUEUE = "IN_QUEUE"
    PRINTING = "PRINTING"
    COMPLETED = "COMPLETED"
    CANCELED = "CANCELED"


class Print:
    id: int
    name: str
    fileExtension: str
    slicedFor: Optional[str]
    status: PrintStatus
    userId: str
    contactDetails: str
    notes: str
    timestamp: str


class UserPrint:
    def __init__(self, print: Print):
        self.id = print.id
        self.name = print.name
        self.fileExtension = print.fileExtension
        self.slicedFor = print.slicedFor
        self.status = print.status
        self.contactDetails = print.contactDetails
        self.notes = print.notes
        self.timestamp = print.timestamp
